sequencer/src/block.rs

```diff
@@ -81,45 +69,147 @@ pub struct Metadata {
     /// genesis of the L1, and the L1 has yet to finalize a block. In all other cases it will be
     /// `Some`.
     pub l1_finalized: Option<L1BlockInfo>,
+
+    pub payload_commitment: VidCommitment,
+    pub transactions_root: NMTRoot,
 }
 
-impl Header {
-    pub fn commit(&self) -> Commitment<Block> {
-        RawCommitmentBuilder::new(&Block::tag())
-            .u64_field("timestamp", self.timestamp())
-            .u64_field("l1_head", self.l1_head())
-            .array_field(
-                "l1_finalized",
-                self.l1_finalized()
-                    .iter()
-                    .map(|block| block.commit())
-                    .collect::<Vec<_>>()
-                    .as_slice(),
-            )
+impl Committable for Header {
+    fn commit(&self) -> Commitment<Self> {
+        RawCommitmentBuilder::new(&Self::tag())
+            .u64_field("height", self.height)
+            .u64_field("timestamp", self.timestamp)
+            .u64_field("l1_head", self.l1_head)
+            .optional("l1_finalized", &self.l1_finalized)
+            .constant_str("payload_commitment")
+            .fixed_size_bytes(&self.payload_commitment.into())
             .field("transactions_root", self.transactions_root.commit())
             .finalize()
     }
+
+    fn tag() -> String {
+        "BLOCK".into()
+    }
 }
 
-#[derive(Clone, Debug, Deserialize, Serialize, Hash, PartialEq, Eq)]
-pub struct Block {
-    metadata: Metadata,
+impl Committable for NMTRoot {
```

Comment:
Do we really need to re-commit the NMT root?

Reply:
I guess we could just fixed_size_bytes it in the header commitment. Not sure if there's a good conversion to bytes for that one. Anyways we should be getting rid of the NMT completely soon 🤞 so I'm not too worried about it